notebooks/264-speculative-sampling/264-speculative-sampling.ipynb

```diff
@@ -0,0 +1,492 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #31.            res = target_model(x_draft_target_input, attention_mask=torch.ones(x_draft_target_input.size(), dtype=torch.long), use_cache=False)
</pre><p>Why use_cache=False? You should use target_past_kv for target_model inference based on <code>x_draft_target_input</code>: </p><pre class="ql-syntax" spellcheck="false">if target_past_kv is None:
    x_draft_target_input = torch.cat((x, x_draft), dim=1)
else:
    x_draft_target_input = x_draft
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1410/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>